test/e2e/proxy_test.go

```diff
@@ -68,7 +70,400 @@ func skipIfProxyDisabled(t *testing.T) {
+	})
+
+	require.NoError(t, data.deletePod(testNamespace, agnhost0))
+	require.NoError(t, data.deletePod(testNamespace, agnhost1))
+	createAgnhostPod(t, data, hostAgnhost0, node0, true)
+	createAgnhostPod(t, data, hostAnghost1, node1, true)
+	t.Run("Host Network Endpoints", func(t *testing.T) {
+		nodePortTestCases(t, data, portStrCluster, portStrLocal, node0IP, node1IP, localhostIP, busybox0, busybox1, busybox0IP, busybox1IP, node0, node1, true)
+	})
+}
+
+func nodePortTestCases(t *testing.T, data *TestData, portStrCluster, portStrLocal, node0IP, node1IP, localhostIP, busybox0, busybox1, busybox0IP, busybox1IP, hostname0, hostname1 string, hostNetwork bool) {
+	clusterUrl0 := net.JoinHostPort(node0IP, portStrCluster)
+	clusterUrl1 := net.JoinHostPort(node1IP, portStrCluster)
+	clusterUrlLo := net.JoinHostPort(localhostIP, portStrCluster)
+	localUrl0 := net.JoinHostPort(node0IP, portStrLocal)
+	localUrl1 := net.JoinHostPort(node1IP, portStrLocal)
+	localUrlLo := net.JoinHostPort(localhostIP, portStrLocal)
+
+	t.Run("ExternalTrafficPolicy:Cluster/Client:Remote", func(t *testing.T) {
+		testNodePortClusterFromRemote(t, data, clusterUrl0, clusterUrl0)
+	})
+	t.Run("ExternalTrafficPolicy:Cluster/Client:Node", func(t *testing.T) {
+		testNodePortClusterFromNode(t, data, clusterUrl0, clusterUrl1, clusterUrlLo)
+	})
+	t.Run("ExternalTrafficPolicy:Cluster/Client:Pod", func(t *testing.T) {
+		testNodePortClusterFromPod(t, data, clusterUrl0, clusterUrl0, busybox0, busybox1)
+	})
+	t.Run("ExternalTrafficPolicy:Local/Client:Remote", func(t *testing.T) {
+		if hostNetwork {
+			t.Skipf("Skip this test as Endpoint is on host network")
+		}
+		testNodePortLocalFromRemote(t, data, localUrl0, localUrl1, node0IP, node1IP, hostname0, hostname1)
+	})
+	t.Run("ExternalTrafficPolicy:Local/Client:Node", func(t *testing.T) {
+		testNodePortLocalFromNode(t, data, localUrl0, localUrl1, localUrlLo, hostname0, hostname1)
+	})
+	t.Run("ExternalTrafficPolicy:Local/Client:Pod", func(t *testing.T) {
+		testNodePortLocalFromPod(t, data, localUrl0, localUrl1, busybox0, busybox1, busybox0IP, busybox1IP, hostname0, hostname1)
+	})
+}
+
+func createAgnhostPod(t *testing.T, data *TestData, podName string, node string, hostNetwork bool) {
+	args := []string{"netexec", "--http-port=8080"}
+	ports := []corev1.ContainerPort{
+		{
+			Name:          "http",
+			ContainerPort: 8080,
+			Protocol:      corev1.ProtocolTCP,
+		},
+	}
+	require.NoError(t, data.createPodOnNode(podName, testNamespace, node, agnhostImage, []string{}, args, nil, ports, hostNetwork, nil))
+	_, err := data.podWaitForIPs(defaultTimeout, podName, testNamespace)
+	require.NoError(t, err)
+	require.NoError(t, data.podWaitForRunning(defaultTimeout, podName, testNamespace))
+}
+
+func createTestClientPod(t *testing.T, data *TestData, client string, node string) (string, string) {
```

Comment:
Can we reuse `createAndWaitForPod`? Its usage is quite simple and flexible, for example:
```
	clientName, _, cleanupFunc := createAndWaitForPod(t, data, data.createBusyboxPodOnNode, "test-client-", "", testNamespace)
	defer cleanupFunc()
```
I think tests that require `createAgnhostPod` can reuse it too.